lib/internal/modules/esm/import_map.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+const { isURL, URL } = require('internal/url');
+const { ObjectEntries, ObjectKeys, SafeMap, ArrayIsArray } = primordials;
+const { codes: { ERR_INVALID_IMPORT_MAP } } = require('internal/errors');
+
+class ImportMap {
+  #baseURL;
+  imports = new SafeMap();
+  scopes = new SafeMap();
```

Comment:
is this class exposed to user code? if so these need to either be private fields or normal Maps, to avoid exposing SafeMap.

Reply:
It's exposed to hook authors, via the context IIUC. On that point, are we OK with hook authors manipulating the import map?

Reply:
Even if hook authors could manipulate the map we’d still not want to expose SafeMap to them; i like making them private.